engines/wintermute/base/gfx/opengl/mesh3ds_opengl_shader.cpp

```diff
@@ -0,0 +1,71 @@
+/* ResidualVM - A 3D game interpreter
+ *
+ * ResidualVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "common/file.h"
+#include "engines/wintermute/base/gfx/opengl/loader3ds.h"
+#include "engines/wintermute/base/gfx/opengl/mesh3ds_opengl_shader.h"
+#include "engines/wintermute/wintypes.h"
+#include "math/vector3d.h"
+
+namespace Wintermute {
+
+Mesh3DSOpenGLShader::Mesh3DSOpenGLShader(OpenGL::Shader *shader) : _shader(shader) {
+}
+
+Mesh3DSOpenGLShader::~Mesh3DSOpenGLShader() {
+	glDeleteBuffers(1, &_vertexBuffer);
```

Comment:
glDeleteBuffers does nothing when the value passed is 0. The constructor does not guarantee any value, so a trivially constructed instance would possibly pass garbage here. This can be fixed by initializing both members in the constructor.

Reply:
The buffers are generated in the constructor now.